### PR TITLE
`docker run/start/exec` flags catch up

### PR DESCRIPTION
**Pull request by bjaglin**
* `docker run` 1.10 flags (I left `ip` / `ip6` out since it is addressed by https://github.com/michaelsauter/crane/pull/270)
* `docker start/exec` catchup
* fallback on run params for flags common to start & run and exec & run

### CODE REVIEW

**Comment by bjaglin** [comment]
I left `ip` / `ip6` out since it will be addressed by https://github.com/michaelsauter/crane/pull/270

File: crane/container.go [hunk 1]
```diff
@@ -973,7 +1126,14 @@ func (c *container) Start(excluded []string) {
 			if c.StartParams().Attach {
 				args = append(args, "--attach")
 			}
-			if c.StartParams().Interactive {
+			if len(c.StartParams().DetachKeys()) > 0 {
+				args = append(args, "--detach-keys", c.StartParams().DetachKeys())
+			} else {
```
Comment (michaelsauter) [inline crane/container.go]: I'd find `else if` on one line a bit more readable.

**Comment by michaelsauter** [comment]
Whoa. This is awesome! :+1: from my side

File: crane/container.go (hunk 1 above)
Comment (bjaglin) [inline crane/container.go]: indeed - fixed, amended & force-pushed

File: crane/container.go [hunk 2]
```diff
@@ -1035,12 +1193,28 @@ func (c *container) Exec(cmds []string) {
...
-		if c.ExecParams().Interactive {
+		if c.ExecParams().Detach || c.RunParams().Detach {
+			args = append(args, "--detach")
+		}
+		if len(c.ExecParams().DetachKeys()) > 0 {
+			args = append(args, "--detach-keys", c.ExecParams().DetachKeys())
+		} else if len(c.RunParams().DetachKeys()) > 0 {
+			args = append(args, "--detach-keys", c.RunParams().DetachKeys())
+		}
+		if c.ExecParams().Privileged || c.RunParams().Privileged {
```
Comment (bjaglin) [inline crane/container.go]: That actually hinders the case where the user would want a `true` value in the run block, but an explicit `false` value in the `exec` one. The fix would be to have a `OptBool` but I guess it's not worth it....

Comment (michaelsauter) [inline crane/container.go]: Yea ... I think this is such an edge case that it's not worth it. I'm fine with the proposed behaviour.

**Comment by bjaglin** [comment]
rebased against master

**Comment by michaelsauter** [comment]
Cool. Feel free to merge!

**Pull request merged**